salt/_beacons/metalk8s_kubeconfig_info.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+import logging
+
+from salt.ext.six.moves import map as _map
+
+__virtualname__ = 'metalk8s_kubeconfig_info'
+
+DEFAULT_NOTIFY_DAYS = 15
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def validate(config):
+    error_msg = "Configuration for {0} beacon is invalid".format(
+        __virtualname__
+    )
+
+    if not isinstance(config, list):
+        return False, "{0}: it must be a list".format(error_msg)
+
+    _config = {}
+    list(_map(_config.update, config))
+
+    if "files" not in config:
+        return False, "{0}: it must include 'files' key".format(error_msg)
+
+    if not isinstance(_config["files"], list):
+        return False, "{0}: 'files' key must be a list".format(error_msg)
+
+    mandatory_keys = {"name", "user"}
+    for kubeconfig in _config["files"]:
+        if not isinstance(kubeconfig, dict):
+            return (
+                False,
+                "{0}: elements in 'files' must be of type dict".format(
+                    error_msg
+                )
+            )
+
+        missing_keys = mandatory_keys - set(kubeconfig.keys())
+        if missing_keys:
+            return (
+                False,
+                "{0}: missing key(s) in 'files' element: {1}".format(
+                    error_msg, ", ".join(missing_keys)
+                )
+            )
+
+    return True, "Valid beacon configuration"
+
+
+def beacon(config):
+    """
+    Watch the configured kubeconfig list
+    """
+    ret = []
+
+    _config = {}
+    list(_map(_config.update, config))
+
+    ca_minion = __pillar__['metalk8s']['ca']['minion']
+    control_plane_ip = __grains__['metalk8s']['control_plane_ip']
+    apiserver = "https://{0}:6443".format(control_plane_ip)
+    b64_ca_cert = __salt__['mine.get'](
+        ca_minion, 'kubernetes_root_ca_b64'
+    )[ca_minion]
+
+    notify_days = _config.get('notify_days', DEFAULT_NOTIFY_DAYS)
+
+    for kubeconfig in _config['files']:
+        remaining_days = kubeconfig.get('notify_days', notify_days)
+        if not __salt__['metalk8s_kubeconfig.validate'](
+                kubeconfig['name'], b64_ca_cert, apiserver,
+                kubeconfig['user'], remaining_days):
+            log.info("kubeconfig %s is out of date", kubeconfig['name'])
```

Comment:
"out of date" is a bit strong, given the mechanism checks "in advance". Maybe something like:

```suggestion
            log.info("kubeconfig %s will expire in less than %d days",
                     kubeconfig['name'], remaining_days)
```

Reply:
However, the beacon actually checks more than the expiry date... And we don't get exactly why the file needs to be regenerated. Maybe something to improve in the future.

Reply:
Yes, maybe not talking about the expiration date is better.
Just tell that the kubeconfig needs to be regenerated.